pkg/util/querylimits/propagation.go

```diff
@@ -22,11 +22,11 @@ const (
 // NOTE: we use custom `model.Duration` instead of standard `time.Duration` because,
 // to support user-friendly duration format (e.g: "1h30m45s") in JSON value.
 type QueryLimits struct {
-	MaxQueryLength          model.Duration   `json:"maxQueryLength,omitempty"`
+	MaxQueryLength          model.Duration   `json:"maxQueryTimeRange,omitempty"`
 	MaxQueryRange           model.Duration   `json:"maxQueryInterval,omitempty"`
 	MaxQueryLookback        model.Duration   `json:"maxQueryLookback,omitempty"`
 	MaxEntriesLimitPerQuery int              `json:"maxEntriesLimitPerQuery,omitempty"`
-	QueryTimeout            model.Duration   `json:"queryTimeout,omitempty"`
+	QueryTimeout            model.Duration   `json:"maxQueryTime,omitempty"`
```

Comment:
hmm that's a breaking change, I'd postpone it until 3.x but in case you think we should have this, make sure to have it on the upgrading guide and changelog

Reply:
It's not really a breaking change because we didn't release/advertise the per-request limit API.

Reply:
aháaa there you go. thank you.